### PR TITLE
adding initial set of scripts PPM-192

### PR DESCRIPTION
**Pull request by IsaacSugden**
Chris Kingsbury's script for exposing data from the Mercury dropdown

### CODE REVIEW

File: scripts/show_semiconductor_properties/show_semiconductor_properties.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import os
+from io import BytesIO
+import base64
+import json
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+
+README_LINK = f"""{Path(os.path.dirname(__file__)) / "ReadMe.md"}"""
+README_LINK = "https://downloads.ccdc.cam.ac.uk/documentation/API/descriptive_docs/predicted_properties.html"
```
Comment (siong-chin) [inline scripts/show_semiconductor_properties/show_semiconductor_properties.py:25]: Which one of these is the real readme link?

File: scripts/show_semiconductor_properties/show_semiconductor_properties.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    else:
+        return fig
+
+
+def write_descs_report(settings=default_settings):
+    interface = ApplicationInterface(parse_commandline=False)
+    interface.parse_commandline()
+    semiconductor_entry_reader = ccdc.io.EntryReader("CSD")
+    entry = semiconductor_entry_reader.entry(interface.identifier)
```
Comment (siong-chin) [inline scripts/show_semiconductor_properties/show_semiconductor_properties.py:93]: Some code meant to be deleted?

**Review by siong-chin — APPROVED** [review]
Looks OK. I assume you have tested it?

Is it meant to be included in the Mercury distribution? You know what to do if so.

**Comment by IsaacSugden** [comment]
> Looks OK. I assume you have tested it?
> 
> Is it meant to be included in the Mercury distribution? You know what to do if so.

yeah I've tested on 1) entries with all data, 2) entries with some data and 3) entries without data and the behaviour is as expected. We'd said we would put it on the ccdc-opensource repo because it's expected to be a community project, which Matt was happy with

File: scripts/show_semiconductor_properties/show_semiconductor_properties.py (hunk 2 above)
Comment (IsaacSugden) [reply scripts/show_semiconductor_properties/show_semiconductor_properties.py:93]: is there an easier way of doing this then? do you mean the parse_comand line bits or the entry bits?

Comment (siong-chin) [reply scripts/show_semiconductor_properties/show_semiconductor_properties.py:93]: The entry bit. 

You have two consecutive lines setting `entry` so it looks the lines 92 and 93 are just for testing?

Comment (IsaacSugden) [reply scripts/show_semiconductor_properties/show_semiconductor_properties.py:93]: ok understood, yeah, I'll delete

**Pull request merged**